docs/spark/how-to-guides/dotnet-interactive-udf-issue.md

```diff
@@ -0,0 +1,65 @@
+We will be using Jupyter Notebooks to illustrate how to use UDFs in an interactive experience.
+
+## UDF serialization in .NET Interactive
+
+In the interactive experience, a cell can be thought of as the code snippet being submitted as part of one iteration of the [REPL](https://en.wikipedia.org/wiki/Read%E2%80%93eval%E2%80%93print_loop). For example, let's take a look at the following notebook to understand what that means:
+
+![Jupyter Notebook cells](./media/dotnet-interactive/dotnet-interactive-cells.png)
+
+Each of those blocks marked by the red arrow is one cell, or one submission of code to Spark. .NET Interactive wraps each of these cells with a wrapper class of the cell submission number, to uniquely identify each cell being submitted. Now as explained in detail in [this guide](udf-guide.md), when a UDF that references a custom object is being serialized its target is also picked up for serialization, which in the case of .NET Interactive gets wrapped by the wrapper class of the cell where the custom object is defined.
+Now let's see how that affects our UDF definition in the notebook:
+
+![UDF serialization error](./media/dotnet-interactive/udf-serialization-error.png)
+
+As can be seen in the case of `udf2_fails`, we see the error message which says Type `Submission#7` is not marked as serializable, this is because how .NET Interactive works is it wraps every object defined in a cell with its `Submission#` class which is generated on the fly and hence is not marked as `Serializable`, hence the error.
+For this reason, it is required that a UDF referencing a custom object in it, is defined in the same cell as that object.
+
+## Calling a UDF on a DataFrame
+
+When calling a previously defined UDF on a `DataFrame` it is important to make sure the UDF is defined in a previously submitted cell, before calling it.
+For example, let's use the UDF shown in the previous example and call it on a DataFrame:
+
+![working UDF call](./media/dotnet-interactive/udf_works.png)
+
+Now let's see what happens if we call the UDF in the same cell where it is defined.
+
+![failing UDF call](./media/dotnet-interactive/udf_fails.png)
+
+The above highlighted error is because the UDF assemblies need to first be compiled and shipped to the workers before it can be called on a DataFrame.
+
+## Why Broadcast Variables don't work with .NET Interactive
+
+The above explained behavior is the reason why Broadcast Variables don't work with .NET Interactive. It is a good idea to go through [this guide on Broadcast Variables](broadcast-guide.md) to get a deeper understanding of what Broadcast variables are and how to use them. The reason Broadcast Variables don't work with interactive scenarios is because of .NET interactive's design of appending each object defined in a cell with it's cell submission class, which since is not marked serialzizable, fails with the same exception as shown previously.
```

Comment:
```suggestion
For the reasons explained above, broadcast variables don't work with .NET Interactive. It is a good idea to go through [this guide on broadcast variables](broadcast-guide.md) to get a deeper understanding of what broadcast variables are and how to use them. The reason broadcast variables don't work with interactive scenarios is because of .NET interactive's design of appending each object defined in a cell with it's cell submission class, which since is not marked serializable, fails with the same exception as shown previously.
```